lib/roo_on_rails/checks/base.rb

```diff
@@ -0,0 +1,72 @@
+require 'roo_on_rails/checks'
+require 'roo_on_rails/checks/helpers'
+
+module RooOnRails
+  module Checks
+    class Base
+      include Helpers
+
+      def self.run(**options)
+        new(**options).run
+      end
+
+      def initialize(fix: false, state: nil)
+        @fix = fix
+        @state = state
+      end
+
+      def run
+        say _intro
+        _call
+      rescue Failure
+        raise unless @fix
+        say "\t· attempting to fix", %i[yellow]
+        _fix
+        say "\t· re-checking", %i[yellow]
+        _call
+      end
+
+      protected
+
+      def _intro
+        self.class.name
+      end
+
+      def _call
+        _fail "this check wasn't implemented"
+      end
+
+      def _fix
+        _fail "can't fix this on my own"
+      end
+
+      def _state
+        @state
+      end
+
+      def _run(cmd)
+        result = Bundler.with_clean_env { %x{#{cmd}} }
+        return [$?.success?, result]
+      end
+
+      def _run!(cmd)
+        Bundler.with_clean_env { system(cmd) }
+        raise CommandFailed.new(cmd) unless $?.success?
```

Comment:
The status code? Yeah, possibly. Although I haven't needed it yet.
Also, `_run!` can't use `_run` because the former calls to `system` (doesn't capture STDOUT) and the latter calls backtick (does capture).